src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Figure figure;
+
+    public Figure getRandomFigure() {
+        GeometricFigure geometricFigure = new GeometricFigureSupplier().getRandomFigure();
```

Comment:
Creating a new instance of 'GeometricFigureSupplier' every time 'getRandomFigure' is called is inefficient. Consider creating a single instance of 'GeometricFigureSupplier' and reusing it.